docs/how-to/use-existing-private-key.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Use an existing private key
+description: How to start an IPFS node with a pre-generated private key.
+---
+
+# Use an existing private key
+
+There are two locations in which IPFS stores private keys:
+
+1. In the config file (`~/.ipfs/config`), following the [config specification](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#identityprivkey) to store the default private key.
+2. In the Keystore (`~/.ipfs/keystore`), following the [fs-repo specification](https://github.com/ipfs/specs/blob/master/REPO_FS.md) to store all other private keys.
+
+Private keys are handled programmatically through code or via the CLI. Here we will show you how to use your existing private keys using both methods.
+
+## CLI (`go-ipfs`)
+
+There's no way of directly initializing an IPFS node using your private keys from IPFS's Go CLI at this moment. To learn more and look for possible workarounds, see the discussion on issue [ipfs/go-ipfs#4240](https://github.com/ipfs/go-ipfs/issues/4240).
+
+You can, however, import private keys into the Keystore:
+
+```sh
+ipfs key import CustomKeyName ./CustomKeyFile
+
+> k51qzi5uqu5dlli90qkt7xcnnjsd6jkn5rbdg7scuv6leozp1ecokewk5rzbeh
+```
+
+The content of the file holding the private key you want to import into the Keystore needs to be in the correct format to be valid.
+The format of Keystore files is outlined in the [fs-repo](https://github.com/ipfs/specs/blob/master/REPO_FS.md#keystore) specification.
+
+For more information on the CLI commands, you can check the help option for the specific command:
+
+```sh
+ipfs COMMAND --help
+```
+
+## CLI - JS (`js-ipfs`)
+
+You can provide the private key inline as a string when initializing (see `--help` for info on valid formats):
+
+```sh
+jsipfs init --private-key 'my-private-key'
+```
+
+You can also import private keys into the Keystore:
```

Comment:
```suggestion
You can also import private keys into the keystore:
```

Reply:
Same as [above](https://github.com/ipfs/ipfs-docs/pull/568#discussion_r545996569).

Reply:
Same as [above](https://github.com/ipfs/ipfs-docs/pull/568#discussion_r545996569).

Reply:
```suggestion
You can also import private keys into the IPFS keystore:
```

As above, since this is referring to an IPFS specific keystore, we should point this out.